commit 513c131c6309712a51502870b041f45b4bd6a6d4
Author: Felix Fietkau <nbd@nbd.name>
Date:   Fri May 17 13:51:22 2024 +0200

    mt76: mt7603: fix mixed declarations and code
    
    Move the qid variable declaration further up
    
    Fixes: b473c0e47f04 ("wifi: mt76: mt7603: fix tx queue of loopback packets")
    Signed-off-by: Felix Fietkau <nbd@nbd.name>

diff --git a/mt7603/dma.c b/mt7603/dma.c
index 14304b06..e238161d 100644
--- a/mt7603/dma.c
+++ b/mt7603/dma.c
@@ -29,7 +29,7 @@ mt7603_rx_loopback_skb(struct mt7603_dev *dev, struct sk_buff *skb)
 	struct ieee80211_sta *sta;
 	struct mt7603_sta *msta;
 	struct mt76_wcid *wcid;
-	u8 tid = 0, hwq = 0;
+	u8 qid, tid = 0, hwq = 0;
 	void *priv;
 	int idx;
 	u32 val;
@@ -57,7 +57,7 @@ mt7603_rx_loopback_skb(struct mt7603_dev *dev, struct sk_buff *skb)
 	if (ieee80211_is_data_qos(hdr->frame_control)) {
 		tid = *ieee80211_get_qos_ctl(hdr) &
 			 IEEE80211_QOS_CTL_TAG1D_MASK;
-		u8 qid = tid_to_ac[tid];
+		qid = tid_to_ac[tid];
 		hwq = wmm_queue_map[qid];
 		skb_set_queue_mapping(skb, qid);
 	} else if (ieee80211_is_data(hdr->frame_control)) {
